Automattic/wp-calypso
# Components: Remove unused disconnect button and dialog components

#### Changes proposed in this Pull Request

* Remove unused `DisconnectJetpackButton` component, along with related unnecessary styles and tests.
* Remove unused `DisconnectJetpackDialog` component, as its last usage was in `DisconnectJetpackButton`. 

Related PRs:
* Removing the last `DisconnectJetpackDialog` usage: #19509 (@ockham I realize it's been a while since you worked on this 😅 )
* Removing the last `DisconnectJetpackButton` usage: #22362.

#### Testing instructions

* Checkout this branch.
* Verify http://calypso.localhost:3000/devdocs/blocks still loads well.
* Smoke test the plugins section for a Jetpack site.
* Verify the removed components are indeed not used anywhere any more.
* Verify tests still pass.
